docs/design/node_taint_eviction.md

```diff
@@ -2,36 +2,42 @@
 
 ## Current Implementation
+The operator does not have this feature. It does not have access to list Node information. It does not consider taint-related events in Pod's condition.
 
-We don't have a current implementation of this and the operator is currently not able to access any node information.
-The closest implementation to this is the [Replace Failed ProcessGroups reconciler](https://github.com/FoundationDB/fdb-kubernetes-operator/blob/main/controllers/replace_failed_process_groups.go), which could be extended for this.
+The operator does replace failed ProcessGroups in the [Replace Failed ProcessGroups reconciler](https://github.com/FoundationDB/fdb-kubernetes-operator/blob/main/controllers/replace_failed_process_groups.go). It can extend it to support this feature.
 
 ## Proposed Design
 
-We have to make sure that this feature can be enabled and disabled, e.g. if someone wants to run the operator inside a cluster without node access.
-The `AutomaticReplacementOptions` will be extended to support the configuration of replacements for Pods that run on tainted nodes.
-The new field will be called `taints` and will contain a list, the default would be an empty list, of taint configurations.
-That configuration will include the taint key, this has to be an exact match or a wildcard `*`, and the duration this taint must be present on the node.
-The `durationInSeconds` allow to specify how sensitive the operator should be to taints.
-If a wildcard is specified as key this will be used as default value, except there is an exact match defined for a taint key.
+### Design Requirements
+**Requirement 1.** The feature can be enabled and disabled by users, because (1) when someone wants to run the operator inside a k8s cluster without Node access, they should be able to bypass this feature; (2) when we notice problems during roll-out of this feature, we need to disable it online.
+
+The feature can be disabled by option `automationOptions.replacements.taintReplacementOptions = {}`.
+
+**Requirement 2.** The feature allows users to configure a set of specific taint keys to detect and replace. This is useful in a multi-tenant k8s environment where FDB operator only wants to react to a subset of taints used in the k8s cluster.
+
+Users can configure the taint feature in the new field `taints` inside `automaticReplacementOptions`. The `taints` field is a list of `(key, durationInSeconds)` pairs, which defines operator's behavior on the taint key.
+
+Users can use `automaticReplacementOptions.taintReplacementTimeSeconds` to control how fast operator should replace Pods on tainted Nodes. Users can ask operator to replace Pods on tainted Nodes much faster than the normal replacement by setting a smaller value of `taintReplacementTimeSeconds`.
```

Comment:
`Users can ask operator to replace Pods on tainted Nodes`, how does a user "asks" the operator to replace Pods on a tainted node? Can we stick to a more technical language? I would prefer the word "configure" and not "ask".